main.go

```diff
@@ -38,7 +144,175 @@ func httpHandler(w http.ResponseWriter, r *http.Request) {
 			" Please check back again later.")); err != nil {
 			log.Printf("error during Write: %v", err)
 		}
+	case http.MethodPost:
+		http.HandleFunc("/verifyProject", verifySignature)
 	default:
 		http.Error(w, "only GET method is allowed", http.StatusMethodNotAllowed)
 	}
 }
+
+func verifyScorecardWorkflow(workflowContent string) bool {
+	// Verify workflow contents using actionlint.
+	workflow, lintErrs := actionlint.Parse([]byte(workflowContent))
+	if lintErrs != nil {
+		return false
+	}
+
+	// Extract main job
+	jobs := workflow.Jobs
+	if len(jobs) != 1 {
+		return false
+	}
+	analysisJob := jobs["analysis"]
+	if analysisJob == nil {
+		return false
+	}
+
+	// Verify that there is no container or services.
+	if analysisJob.Container != nil || len(analysisJob.Services) > 0 {
+		return false
+	}
+
+	// Verify that the workflow runs on ubuntu-latest.
+	if len(analysisJob.RunsOn.Labels) != 1 || analysisJob.RunsOn.Labels[0].Value != "ubuntu-latest" {
+		return false
+	}
+
+	// Verify that there are no env vars set.
+	if analysisJob.Env != nil {
+		return false
+	}
+
+	// Get steps in job.
+	steps := analysisJob.Steps
+
+	if steps == nil || len(steps) > 4 {
+		return false
+	}
+
+	// Verify that steps are valid (checkout, scorecard-action, upload-artifact, upload-sarif).
+	for _, step := range steps {
+		stepName := step.Exec.(*actionlint.ExecAction).Uses.Value
+		stepName = stepName[:strings.Index(stepName, "@")] // get rid of commit sha.
```

Comment:
verify that it's present. It's allowed to not use it, but we don't want to encourage this behavior.
Shall we also verify that it's pinned by hash or not? Not exactly necessary, so up to you